pkg/ccl/changefeedccl/kvevent/blocking_buffer.go

```diff
@@ -39,13 +38,23 @@ type blockingBuffer struct {
 // It will grow the bound account to buffer more messages but will block if it
 // runs out of space. If ever any entry exceeds the allocatable size of the
 // account, an error will be returned when attempting to buffer it.
-func NewMemBuffer(acc mon.BoundAccount, metrics *Metrics) Buffer {
+func NewMemBuffer(acc mon.BoundAccount, metrics *Metrics, opts ...quotapool.Option) Buffer {
 	bb := &blockingBuffer{
 		signalCh: make(chan struct{}, 1),
 	}
 	bb.acc = acc
 	bb.metrics = metrics
-	bb.qp = quotapool.New("changefeed", &bb.blockingBufferQuotaPool)
+
+	opts = append(opts,
+		quotapool.OnAcquisition(func(
+			ctx context.Context, poolName string, r quotapool.Request, start time.Time,
+		) {
+			if r.(*bufferEntry).wasBlocked {
+				metrics.BufferPushbackNanos.Inc(timeutil.Since(start).Nanoseconds())
+			}
+		}))
```

Comment:
Consider this change and using `OnWait`, the `quotapool` is already doing all the bookkeeping you need, shame to duplicate it here. 

```go
diff --git a/pkg/kv/kvserver/tenantrate/limiter.go b/pkg/kv/kvserver/tenantrate/limiter.go
index 1e117acc54..7b07a7aff7 100644
--- a/pkg/kv/kvserver/tenantrate/limiter.go
+++ b/pkg/kv/kvserver/tenantrate/limiter.go
@@ -91,7 +91,7 @@ func (rl *limiter) init(
                func(ctx context.Context, poolName string, r quotapool.Request) {
                        rl.metrics.currentBlocked.Inc(1)
                },
-               func(ctx context.Context, poolName string, r quotapool.Request) {
+               func(ctx context.Context, poolName string, r quotapool.Request, _ time.Time) {
                        rl.metrics.currentBlocked.Dec(1)
                },
        ))
diff --git a/pkg/util/quotapool/config.go b/pkg/util/quotapool/config.go
index 128d70160d..21f58c022b 100644
--- a/pkg/util/quotapool/config.go
+++ b/pkg/util/quotapool/config.go
@@ -37,16 +37,16 @@ func OnAcquisition(f AcquisitionFunc) Option {
        })
 }
 
-// OnWaitFunc is the prototype for functions called to notify the start or
-// finish of a waiting period when a request is blocked.
-type OnWaitFunc func(
+// OnWaitStartFunc is the prototype for functions called to notify the start
+// of a waiting period when a request is blocked.
+type OnWaitStartFunc func(
        ctx context.Context, poolName string, r Request,
 )
 
 // OnWait creates an Option to configure two callbacks which are called when a
 // request blocks and has to wait for quota (at the start and end of the
 // wait).
-func OnWait(onStart, onFinish OnWaitFunc) Option {
+func OnWait(onStart OnWaitStartFunc, onFinish AcquisitionFunc) Option {
        return optionFunc(func(cfg *config) {
                cfg.onWaitStart = onStart
                cfg.onWaitFinish = onFinish
@@ -112,13 +112,14 @@ func WithMinimumWait(duration time.Duration) Option {
 }
 
 type config struct {
-       onAcquisition             AcquisitionFunc
-       onSlowAcquisition         SlowAcquisitionFunc
-       onWaitStart, onWaitFinish OnWaitFunc
-       slowAcquisitionThreshold  time.Duration
-       timeSource                timeutil.TimeSource
-       closer                    <-chan struct{}
-       minimumWait               time.Duration
+       onAcquisition            AcquisitionFunc
+       onSlowAcquisition        SlowAcquisitionFunc
+       onWaitStart              OnWaitStartFunc
+       onWaitFinish             AcquisitionFunc
+       slowAcquisitionThreshold time.Duration
+       timeSource               timeutil.TimeSource
+       closer                   <-chan struct{}
+       minimumWait              time.Duration
 }
 
 var defaultConfig = config{
diff --git a/pkg/util/quotapool/quotapool.go b/pkg/util/quotapool/quotapool.go
index 7be0319eb7..ec9b24bc5e 100644
--- a/pkg/util/quotapool/quotapool.go
+++ b/pkg/util/quotapool/quotapool.go
@@ -220,7 +220,7 @@ func (qp *AbstractPool) Acquire(ctx context.Context, r Request) (err error) {
                qp.config.onWaitStart(ctx, qp.name, r)
        }
        if qp.config.onWaitFinish != nil {
-               defer qp.config.onWaitFinish(ctx, qp.name, r)
+               defer qp.config.onWaitFinish(ctx, qp.name, r, start)
        }
 
        // Set up the infrastructure to report slow requests.
```